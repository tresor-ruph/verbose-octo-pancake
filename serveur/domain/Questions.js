const Joi = require('joi')


module.exports = (order, question, image, answer,pollId) => {

  const imageValid = (value, helpers) => {
    if (value === '' || value.length > 6) {
      return value
    } else {
      throw new Error('invalid image')
    }
  }


  const schema = Joi.object({
    order: Joi.number(),
    question: Joi.string().min(3).required(),
    image: Joi.custom(imageValid, 'custom validation'),
    answer: Joi.number().required(),
    pollId: Joi.string().min(10).required(),


  })

  const { error, value } = schema.validate({order, question, image, answer,pollId })
  if (error) {
    return { error: error.details }
  }
  return { value: value }

}