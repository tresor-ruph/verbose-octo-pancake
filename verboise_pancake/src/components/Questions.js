import React, { useState, useEffect } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

import './question.scss'
const Questions = (props) => {
    const [count, setCount] = useState(2)
    const [inputField, setInputField] = useState([1, 2])

    const addField = () => {
        if (inputField.length >= 6) {
            return
        }
        inputField.push(count + 1)
        setCount(prev => prev + 1)
    }

    const rmvField = (x,y) => {
        let newArr = []
        newArr = inputField.filter(elt => elt != x)
        props.removeFieldArray(x, y)
        setInputField(newArr)
    }

    const displayElts = () => {
        let arr = []
        Array.from(inputField).forEach(elt => {
            arr.push(
                <div key={elt} ><input type='text' className='ques-opt' placeholder={'option'+elt} onChange={(event) =>props.handleOptionChange(props.questionCount,elt,event)} />
                { ![1,2].includes(elt)  &&  <FontAwesomeIcon onClick={() => rmvField(elt, props.questionCount)} icon='minus-square' color='red' />}
                </div>
            )
        })
        return arr
    }

    const revealOptions = (id) => {
        let optionsDiv = document.getElementsByClassName('options-list')
        Array.from(optionsDiv).forEach(elt => {
            elt.style.display ='none'
          });
        document.getElementById('option'+id).style.display = 'block'
        let questDiv = document.getElementsByClassName('edit-quests')
        Array.from(questDiv).forEach(elt => {
          elt.style.display='inline'
        })
    }
    
    return (
        <div key={props.questionKey}>
            <input type='text' className = 'new-quest' placeholder={'Question '+props.questionKey} onChange={(event) => props.handleQuestions(props.questionKey, event)} />
            <FontAwesomeIcon onClick={() => revealOptions(props.questionKey)} icon='pencil-alt' color='blue' size='sm'  style ={{display:'none', marginRight: '10px'}} className='edit-quests' id={'edit-quest'+props.questionKey} />     
            <FontAwesomeIcon onClick={() => props.rmvQuestion(props.questionCount, props.questionKey)} icon='trash-alt' color='red' size='sm' />
            <br />
            <div className={'hide-options options-list'} id={'option'+props.questionKey}>
            {displayElts()}
           
            <div className='add-icon'>
            <FontAwesomeIcon icon="plus-square"  color='green' size='lg' onClick={() => addField()}  />
            </div>
            </div>
        </div>

    )
}
export default Questions