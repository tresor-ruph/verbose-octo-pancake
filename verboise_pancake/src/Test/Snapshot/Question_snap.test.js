import React from 'react'
import renderer from 'react-test-renderer'
import Questions  from './../../components/Questions'

it('renders correctly', () => {
    const QuestionsSnap = renderer.create(<Questions />).toJSON();
    expect(QuestionsSnap).toMatchSnapshot();

});