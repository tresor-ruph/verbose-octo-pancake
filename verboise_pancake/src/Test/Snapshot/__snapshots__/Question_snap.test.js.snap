// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <input
    className="new-quest"
    onChange={[Function]}
    placeholder="Question undefined"
    type="text"
  />
  <br />
  <div
    className="hide-options options-list"
    id="optionundefined"
  >
    <div>
      <input
        className="ques-opt"
        onChange={[Function]}
        placeholder="option1"
        type="text"
      />
    </div>
    <div>
      <input
        className="ques-opt"
        onChange={[Function]}
        placeholder="option2"
        type="text"
      />
    </div>
    <div
      className="add-icon"
    />
  </div>
</div>
`;
